apptools/logger/plugin/logger_plugin.py

```diff
@@ -15,10 +15,16 @@
 """
 
 # Standard library imports.
+import sys
 import logging
+from unittest.mock import MagicMock
 
 # Enthought library imports.
-from envisage.api import ExtensionPoint, Plugin
+try:
+    from envisage.api import ExtensionPoint, Plugin
+except ModuleNotFoundError:
+    sys.modules['envisage.api'] = MagicMock()
+    from envisage.api import ExtensionPoint, Plugin
```

Comment:
This affects production code too and this is going to cause strange behaviour that is much harder to debug than getting an ImportError outright.

Mocking for building documentation should only happen within the documentation build workflow. I'd suggest adding this in `docs/source/conf.py`:
```
# Apptools offers an envisage plugin that requires importing from
# envisage.
try:
    import envisage
except ImportError:
    autodoc_mock_imports = [
        'envisage',
    ]

```
and revert this change in `apptools.logger.plugin.logger_plugin`.